src/mockups/VaultV2.sol

```diff
@@ -312,10 +326,14 @@ contract VaultV2 {
      * or the smaller the on-chain liquidity, the biggert the haircut will be.
      */
     function getCollateralValue(uint256 vaultValue) public view returns (uint256 collateralValue) {
```

Comment:
As agreed, this function should be deleted.
There is no point in pre-calculating the vaultValue if we have to calculate the collateralFactor separately.
Both have to loop over all assets.